### ISSUE TITLE
Relicense with BSD-3-Clause license.

### ISSUE THREAD

**Issue by jemc** [jemc/mare#244]
### TL;DR
If you're a named contributor of code I have pinged in this message, **please leave a comment granting your permission** to relicense this repository.

---

The original license I used when I first started experimenting with this repo was MPLv2, based on thinking that if I started with a sorta-copyleft license I could always loosen it to a more commercial-friendly license like MIT or BSD later.

However, it turns out that my thinking was backwards, because inbound contributions are also considered to be licensed with MPLv2, and thus cannot be loosened to a less restrictive license like MIT or BSD without the explicit permission of contributors (who may not otherwise agree to have their work available with a less restrictive license).

Whereas, if the project was licensed as MIT or BSD, those licenses don't restrict further redistributing/republishing the work under more restrictive licenses like MPLv2, or even GPL if it was desired. However, it's worth noting that, as with any open source relicensing, earlier versions of the work would still be usable under their original license.

I am still by far the primary contributor to this project, but I've had help from some folks who have committed code along the way, and I'd like to seek their permission now to relicense before the contributor list starts growing bigger and a relicense process becomes more unwieldy.

---

After research, I believe that the [BSD-3-Clause license](https://opensource.org/licenses/BSD-3-Clause) is the most appropriate one for this project.

It restricts the end-user in very few ways, and still provides key protections to the copyright holder. It is friendly to use in both commercial projects and in copyleft-leaning communities. I think it's the right choice for a general-purpose programming language.

---

@tonchis @mneumann @teggotic @mfelsche @jasoncarr0 - please reply to this issue ticket with the following message:

---

> I hereby grant Joe Eli McIlvain and [the Savi language project](https://github.com/savi-lang/savi) permission to re-license my contributions under [the BSD-3-Clause license, as documented by the Open Source Initiative](https://opensource.org/licenses/BSD-3-Clause).

**Comment by tonchis** [comment]
I hereby grant Joe Eli McIlvain and the Savi language project permission to re-license my contributions under the BSD-3-Clause license, as documented by the Open Source Initiative.

**Comment by jasoncarr0** [comment]
Whoops, lost track of this.

@jemc I hereby grant Joe Eli McIlvain and [the Savi language project](https://github.com/savi-lang/savi) permission to re-license my contributions under [the BSD-3-Clause license, as documented by the Open Source Initiative](https://opensource.org/licenses/BSD-3-Clause).

That said, I'm not aware of any copyrightable contributions of mine

**Comment by jemc** [comment]
> That said, I'm not aware of any copyrightable contributions of mine

You have some commits on an experimental branch that may eventually make it into the main branch.

**Comment by teggotic** [comment]
I hereby grant Joe Eli McIlvain and [the Savi language project](https://github.com/savi-lang/savi) permission to re-license my contributions under [the BSD-3-Clause license, as documented by the Open Source Initiative](https://opensource.org/licenses/BSD-3-Clause).

**Comment by mfelsche** [comment]
I hereby grant Joe Eli McIlvain and the Savi language project permission to re-license my contributions under the BSD-3-Clause license, as documented by the Open Source Initiative.

**Comment by jemc** [comment]
Great, I now have relicense grant comments from the 5 code contributors I named above.

Just for good measure I'll also do the grant comment myself:

---

I hereby grant Joe Eli McIlvain and the Savi language project permission to re-license my contributions under the BSD-3-Clause license, as documented by the Open Source Initiative. 

---

Now I will proceed with relicensing
